### ISSUE TITLE
Improve connection error messaging

### ISSUE THREAD

**Issue by mark-prins** [openmsupply/mobile#4361]
## Is your feature request related to a problem? Please describe.

Bring the changes from [authentication utilities](https://github.com/sussol/authentication-utilities/issues/7) which will surface a better error than `Unable to connect`

## Describe the solution you'd like

Better errors when unable to connect.

## Implementation

Remove the dependency on `sussol-utilities` and `authentication-utilities` and implement the changes directly in mobile.

**Comment by ArpitaSussol** [comment]
Hi @mark-prins, in the above what need to be compared from old and new connection error message? Or
just need to cross check the error connection message is working as expected or not?

**Comment by mark-prins** [comment]
It would be enough to check that the connection message is working. 
I _think_ in this version I broke the 'site registration' error message, and it was fixed in a later issue. Worth just checking the common errors in the current mobile version I think. 

- site registration doesn't match ( when the hardware ID is different in mSupply )
- login incorrect ( wrong username / password )
- unable to connect ( server offline or tablet has no network connection )
